pkg/content/registry.go

```diff
@@ -39,17 +39,21 @@ type RegistryOptions struct {
 // Registry provides content from a spec-compliant registry. Create an use a new one for each
 // registry with unique configuration of RegistryOptions.
 type Registry struct {
+	Opts docker.ResolverOptions
 	remotes.Resolver
 }
 
 // NewRegistry creates a new Registry store
 func NewRegistry(opts RegistryOptions) (*Registry, error) {
+	r, dopts := newResolver(opts.Username, opts.Password, opts.Insecure, opts.PlainHTTP, opts.Configs...)
+
 	return &Registry{
-		Resolver: newResolver(opts.Username, opts.Password, opts.Insecure, opts.PlainHTTP, opts.Configs...),
+		Resolver: r,
+		Opts:     dopts,
```

Comment:
Why do you need this ResolverOptions? For discoverer?

Reply:
The discoverer still needs to resolve the initial descriptor before it can use the list referrers api. Passing the options here allows me to re-instantiate the resolver's internals.